### DISCUSSION TITLE
how to exclude abiFilters?

### DISCUSSION THREAD

**Discussion by aafnnp** [expo/expo#29750] [Archived: Questions]
![image](https://github.com/expo/expo/assets/12791216/6b78762b-42bb-4f1a-a674-f707cf0f435e)
how to exclude x86 lib?

**Comment by wekor** [comment]
Have you dealt with it?

**Reply by aafnnp** [reply]
no

**Reply by FSpark** [reply]
I tried to modify app/build.gradle` directly and then compile it and found that there is still

**Comment by BabyLy233** [comment]
Okay, now I know how to handle this.

Two solutions:

1. Use `expo-build-properties` [doc](https://docs.expo.dev/versions/latest/sdk/build-properties/#buildpropertieswithbuildpropertiesconfig-props)

Modify `app.json`

```json
      [
        "expo-build-properties",
        {
          "android": {
            "buildArchs": ["armeabi-v7a", "arm64-v8a"]
          }
        }
      ]
```
Then run `npx expo prebuild` to regenerate android folder.

2. If compiling manually

Modify `.\android\gradle.properties`

```properties

# Use this property to specify which architecture you want to build.

# You can also override it from the CLI using

# ./gradlew <task> -PreactNativeArchitectures=x86_64
reactNativeArchitectures=armeabi-v7a,arm64-v8a
```
Then run command `.\gradlew.bat assembleRelease`

**Comment by lukmccall** [comment]
You can also pass the `android.injected.build.abi` flag to your build command if you want to build only one arch. It works for packages that aren't configured correctly to handle `reactNativeArchitectures` flag. 

Exmaple:
```
./gradlew :app:assembleRelease -Pandroid.injected.build.abi=arm64-v8a 
````
